File: app/src/main/java/com/inqint/yarnrequirements/Projects/Scarf.java
package com.inqint.yarnrequirements.Projects;

/**
 * Created by deb on 4/27/16.
 */
public class Scarf extends DimensionProject {


    public Scarf(String name, int thumbImageID, int imageID, Class<?> fragmentClass) {
        super(name, thumbImageID, imageID, fragmentClass);
    }

    /* Make sure units are in SI units, then calculate based on the length & width */
    @Override
    public void calcYarnRequired()
    {
        double siLength = length;
        if (getLengthUnits() != ShortLengthUnits.cm)
        {
            siLength = length * inches2cm;
        }
        double siWidth = width;
        if (getWidthUnits() != ShortLengthUnits.cm)
        {
            siWidth = width * inches2cm;
        }
        super.calcYarnRequired(siLength, siWidth);
    }

}

File: app/src/main/java/com/inqint/yarnrequirements/Projects/Scarf.kt
package com.inqint.yarnrequirements.Projects

/**
 * Created by deb on 4/27/16.
 */
open class Scarf(name: String, nameID: Int, thumbImageID: Int, imageID: Int) :
    DimensionProject(name, nameID, thumbImageID, imageID) {

    /* Make sure units are in SI units, then calculate based on the length & width */
    override fun calcYarnRequired() {
        var siLength = length
        if (lengthUnits !== ShortLengthUnits.cm) {
            siLength = length * inches2cm
        }
        var siWidth = width
        if (widthUnits !== ShortLengthUnits.cm) {
            siWidth = width * inches2cm
        }
        super.calcYarnRequired(siLength, siWidth)
    }

}
